pages/form/WorkOnSiteDays.jsx

```diff
@@ -40,18 +40,28 @@ export default function WorkOnSiteDays() {
   }
 
   // we  pass this function as props to our child component to update Form data and logs
-  const saveDataAndShowLog = () => {
+  const saveDataAndShowLog = (msg="Next button clicked") => {
     saveAnswers();
-    sendLogs(logMessage("Next button clicked"));
+    sendLogs(logMessage(msg));
+  }
+
+  const backButtonOnClickHandler = (msg="Back button clicked") => {
+    saveDataAndShowLog(msg)
+    history.back();
```

Comment:
'Back' `<Button>`'s `onClick` handler.
saves responses, and sends logs, then goes to previous page using browser's `history.back()` function.